### ISSUE TITLE
Add to f-droid

### ISSUE THREAD

**Issue by DamienCassou** [vector-im/riotX-android#428]
Hi,

I would like to try riotX but don't have Play Store or even Google services. Would it be possible to have riotX on f-droid.org?

**Comment by krombel** [comment]
While RiotX is not part of the repository on f-droid.org I can suggest my F-Droid repos which contain the current development builds.
There are two flavors: One which is based on using Google Services but one as well which does not rely on those and would therefore be compatible to f-droid.org.
You might checkout https://fdroid.krombel.de for the links to the flavor you want to try.

**Comment by DamienCassou** [comment]
> While RiotX is not part of the repository on f-droid.org I can suggest my F-Droid repos which contain the current development builds.

Would it be possible to get your repository mentioned in RiotX README file? That way, people know they can trust it.

**Comment by Bubu** [comment]
RfP is open here: https://gitlab.com/fdroid/rfp/issues/1035

Right now this is pulling libraries from two `unkown maven repos` [here](https://github.com/vector-im/riotX-android/blob/develop/build.gradle#L42) and [here](https://github.com/vector-im/riotX-android/blob/develop/build.gradle#L51)

For https://repo.adobe.com/nexus/content/repositories/public/ I have no idea where I'd find the source for that. For the piasy stuff that shoudl really just be published to jcenter instead it's own repo, there's a closed issue about this here: https://github.com/Piasy/BigImageViewer/issues/113

**Comment by neVERberleRfellerER** [comment]
Link to source of diff_match_patch is in pom (https://repo.adobe.com/nexus/content/repositories/public/diff_match_patch/diff_match_patch/current/diff_match_patch-current.pom). But there is a catch: it links to Google Code (http://code.google.com/p/google-diff-match-patch) which redirects to github and import squashed everything into one commit. This is quite problematic, because jar was published in 2013 and it is not possible to get the original source from 2013. Waybackmachine works and shows commit history (https://web.archive.org/web/20160531121855/http://code.google.com:80/p/google-diff-match-patch/source/list) Using commits in my favourite web search engine, I found this:
https://github.com/GerHobbelt/google-diff-match-patch/blob/master/java/name/fraser/neil/plaintext/diff_match_patch.java
which ends with last commit relevant to Java from original Google Code repo. But I guess that this is too weak to be useful.

**Comment by bmarty** [comment]
Waiting for an official F-Droid version of RiotX, feel free to use @krombel 's repositories: https://fdroid.krombel.de/, which provide a nightly build gracefully

**Comment by Bubu** [comment]
@neVERberleRfellerER Thanks for your research work here.
@bmarty or anyone else working on this: Any chance to vendor the diff_match_patch implementation instead of pulling a random jar file from somewhere?

**Comment by MayeulC** [comment]
Couldn't https://github.com/google/diff-match-patch/blob/master/java/tests/name/fraser/neil/plaintext/diff_match_patch_test.java be used, both for vector's build and F-Droid's?

According to the instructions in that file, this would be:

```
git clone https://github.com/google/diff-match-patch.git
cd diff-match-patch.git/java
javac -d classes src/name/fraser/neil/plaintext/diff_match_patch.java tests/name/fraser/neil/plaintext/diff_match_patch_test.java
jar cf diff_match_patch-current.jar classes/name/fraser/neil/plaintext/*.class
```

This can probably be better integrated in build.grafle ([maybe by making that repo a submodule](https://stackoverflow.com/a/29791978))? But I don't know much more about this. I'll see if I can do something.

**Comment by Bubu** [comment]
Blocked by #814

**Comment by Bubu** [comment]
#815 is not a blocker but it's one more thing we'd need to patch out on the f-droid side.

**Comment by ilmaisin** [comment]
It seems to have been added there recently: https://f-droid.org/en/packages/im.vector.riotx/

Should this now be closed?

**Comment by bmarty** [comment]
Yes

**Comment by Miaourt** [comment]
Hi, seems that the F-Droid version is stuck on the 0.13.0 :(

**Comment by rakshazi** [comment]
Hello,
F-droid store does not contain new version of application, could you upload it, please?

**Comment by bmarty** [comment]
Yes, it will be uploaded after the rebranding.

**Comment by heyakyra** [comment]
>Hi, seems that the F-Droid version is stuck on the 0.13.0 :(

Version seems stuck again?
https://github.com/vector-im/element-android/issues/1857#issuecomment-845583353

**Comment by Miaourt** [comment]
F-Droid is at 1.1.7 right now. That's not the latest ?

**Comment by DamienCassou** [comment]
This issue has been closed a long time ago and the original problem has been fixed. I suggest opening a new issue.
